runtime/yaml.cpp

```diff
@@ -0,0 +1,169 @@
+#include <yaml-cpp/yaml.h>
+
+#include "runtime/optional.h"
+#include "runtime/streams.h"
+#include "runtime/critical_section.h"
+
+#include "runtime/yaml.h"
+
+// TODO: add key-arrays support
+void yaml_node_to_mixed(const YAML::Node &node, mixed &data) {
+  data.clear();
+  if (node.IsNull()) {
+    php_warning("Yaml node is null. Skipping it");
+    return;
+  }
+  if (node.IsScalar()) {
+    string string_data = string(node.as<std::string>().c_str());
+    if (string_data.is_int()) {
+      data = string_data.to_int();
+      return;
+    }
+    dl::enter_critical_section();
+    double *float_data = new double;
+    if (string_data.try_to_float(float_data)) {
+      data = *float_data;
+    }
+    else data = string_data;
+    delete float_data;
+    dl::leave_critical_section();
+    return;
+  }
+  if (node.IsSequence()) {
+    for (auto it = node.begin(); it != node.end(); ++it) {
+      mixed data_piece;
+      yaml_node_to_mixed(*it, data_piece);
+      data.push_back(data_piece);
+    }
+  }
+  else if (node.IsMap()) {
+    for (auto it : node) {
```

Comment:
It's better to indicate here that `it` is constant.
And it's also better to take the value by reference, so as not to initialize it with a copy.
```c++
for (const auto &it : node) {
```